contracts/0.4.24/oracle/LidoOracle.sol

```diff
@@ -83,6 +89,10 @@ contract LidoOracle is ILidoOracle, AragonApp {
     bytes32 internal constant MAX_REPORTED_EPOCH_ID_POSITION = keccak256("lido.LidoOracle.maxReportedEpochId");
     /// @dev storage for all gathered from reports data
     mapping(uint256 => EpochData) private gatheredEpochData;
+    /// @dev historic data about 2 last completed reports
+    bytes32 internal constant COMPLETED_REPORT_ID_LAST = bytes32(0);
+    bytes32 internal constant COMPLETED_REPORT_ID_PREV = bytes32(1);
+    mapping(bytes32 => CompletedReport) private lastCompletedReports;
```

Comment:
mapping are inefficient and not work with unstructured storage. better to make it just two constant address structures

Reply:
or get rid of it alltogether

Reply:
Fixed by using 4 plain `uint256` variables. Not so elegant, but I agree, it is the best we can do here.